server/spec/requests/api/v1/mqtt_spec.rb

```diff
@@ -0,0 +1,19 @@
+require 'rails_helper'
+
+RSpec.describe 'Api::V1::MqttController', type: :request do
+  describe 'POST #publish' do
+    it 'publishes light on to the MQTT broker' do
+      client = double(MQTT::Client)
+      allow(MQTT::Client).to receive(:connect).and_yield(client)
```

Comment:
So this is just to make sure that the `MQTT:Client` will yield `client` if the `connect` method is called successfully.

Reply:
Hmmm, I think so! This is how I've come to understand it:
1. A method called `connect` is called on the normal (not stubbed) `MQTT::Client` 
2. When that `connect` method is called, specifically return to me `client`, which is our stubbed version.